functests/cluster_upgrade_test.go

```diff
@@ -0,0 +1,56 @@
+package functests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/openshift/ocs-operator/functests"
+
+	deploymanager "github.com/openshift/ocs-operator/pkg/deploy-manager"
+)
+
+var _ = Describe("Cluster upgrade", func() {
+	flag.Parse()
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+	})
+
+	Describe("ocs", func() {
+
+		AfterEach(func() {
+			By("Uninstall the upgraded cluster")
+			tests.AfterUpgradeTestSuiteCleanup()
+			By("Reinstall a fresh cluster after upgrade")
+			tests.BeforeTestSuiteSetup()
+		})
+
+		Context("upgrade cluster", func() {
+			It("and verify deployment status", func() {
+				if tests.UpgradeToOcsRegistryImage == "" || tests.UpgradeToLocalStorageRegistryImage == "" {
+					Skip("Condition not met for upgrade")
+				}
+				deployManager, err := deploymanager.NewDeployManager()
+				Expect(err).To(BeNil())
+
+				By("Getting the current csv before the upgrade")
+				csv, err := deployManager.GetCsv()
+				Expect(err).To(BeNil())
+				
+				By("Upgrading OCS with OLM")
+				err = deployManager.UpgradeOCSWithOLM(tests.UpgradeToOcsRegistryImage, tests.UpgradeToLocalStorageRegistryImage, tests.UpgradeToOcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				By("Waiting for OCS CatalogSource")
+				err = deployManager.WaitForUpgradeCatalogSource(csv.Name, tests.UpgradeToOcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				// Make sure StorageCluster previously created in the environment is still healthy
+				By("Waiting for StorageCluster")
+				err = deployManager.WaitOnStorageCluster()
+				Expect(err).To(BeNil())
```

Comment:
we need some sort of verification that the update occurred. Maybe cross examine that the CSV is posted and wait on the deployments (ocs-operator, noobaa-operator, rook-ceph-operator) roll over successfully.  Then I suppose we need to verify the CephCluster is updated with the ceph container image in the update as well.

Reply:
In the previous step, we do already check for csv to be posted and succeeded, which has reliably indicated upgrade success/failure so far. I also added checks to verify catalog-source image and ocs-operator deployment image after the upgrade.
Upgrade doesn't necessarily has to have new images for components like ceph. I guess we can introspect the two csvs and decide when to expect new images to be deployed.

Reply:
Added checks to verify the deployment images matches the one specified in the csv for each operators.

Reply:
Added checks to verify deployment images match the ones specified in the csv install spec for each operator.